### PR TITLE
feat: Add unloadAndSavePreload

### PR DESCRIPTION
**Pull request by theodab**
This adds a supplementary new method, unloadAndSavePreload. This method unloads the current asset but creates a PreloadManager object containing the loaded manifest, so the asset can be loaded again more quickly. This can be used for situations where an asset must be unloaded only temporarily; say, you wish to detach the player from the current media element and re-attach it to another, then begin playing the same asset.

### CODE REVIEW

**Comment by theodab** [comment]
This was broken off of #5897, as the feature was not central to preload functionality and could be improved a little.

Possible improvements to make:

- Create a segment prefetch for the preload manager.
- Save the current variant.
- Save the DRM engine, if possible.

File: lib/player.js
Comment (avelad) [inline lib/player.js:1618]: We should save the currentTime and set it as startTime. For live is more complicate because it has to be relative, but it is possible.

Comment (avelad) [reply lib/player.js]: Done!

**Review by avelad — APPROVED** [review]

**Comment by avelad** [comment]
@joeyparrish can you review it? Thanks!

File: lib/player.js
Comment (theodab) [inline lib/player.js:1641]: You will also want to set `this.abrManager_` to null, so that the player doesn't call `this.abrManager_.stop()` while unloading.

File: lib/media/preload_manager.js
Comment (theodab) [inline lib/media/preload_manager.js:375]: This should not set `this.abrManagerEntrusted_`.
That variable means that the lifecycle of the abr manager is in the hands of a player instance. Ideally, when `unloadAndSavePreload` is called, all of the managers passed to it are detangled from the player that made them.

Comment (avelad) [reply lib/media/preload_manager.js]: Done!

**Comment by theodab** [comment]
Okay, I can't approve this, since it's technically my PR, but your additions have my approval now.

**Review by avelad — APPROVED** [review]

**Pull request merged**
